src/main/java/me/rigamortis/seppuku/impl/module/ui/HudEditorModule.java

```diff
@@ -18,44 +20,30 @@
     public final Value<Boolean> blur = new Value("Blur", new String[]{"b"}, "Apply a blur effect to the Hud Editor's background.", true);
     public final Value<Boolean> tooltips = new Value("ToolTips", new String[]{"TT", "Tool"}, "Displays tooltips for modules.", true);
 
-    private boolean open;
-
     public HudEditorModule() {
         super("HudEditor", new String[]{"HudEdit", "HEdit"}, "Displays a menu to modify the hud", "GRAVE", -1, ModuleType.UI);
         this.setHidden(true);
     }
-
+    
     @Override
-    public void onToggle() {
-        super.onToggle();
-        final Minecraft mc = Minecraft.getMinecraft();
-
-        if (mc.world != null) {
-            mc.displayGuiScreen(new GuiHudEditor());
-
-            if(this.blur.getValue()) {
-                if (OpenGlHelper.shadersSupported) {
-                    mc.entityRenderer.loadShader(new ResourceLocation("minecraft", "shaders/post/blur.json"));
-                }
-            }
-            this.open = true;
+    public void onEnable() {
+        if (Minecraft.getMinecraft().world != null) {
+            Seppuku.INSTANCE.getEventManager().addEventListener(this);
         }
     }
-
+    
+    @Override
+    public void onDisable() {}
+    
     @Listener
-    public void displayGui(EventDisplayGui event) {
-        if (Minecraft.getMinecraft().currentScreen instanceof GuiHudEditor) {
-            if (event.getScreen() == null && this.open) {
-                event.setCanceled(true);
-            }
+    public void onPlayerUpdate(EventPlayerUpdate event) {
```

Comment:
Discord message I sent a while ago for reference:

>>> @riga#0278 btw doing `.t HudEditor` doesnt work in seppuku because when you send a chat message the the chat automatically sets the guiscreen to null in order to close the chat screen, and with commands opening a gui will occur before the guichat closes the gui, if that makes sense. 

Here's how I got around it, if you want to do the same for seppuku:
```Java
@Override
    public boolean execute(String[] args)
    {
        toOpen = true;
        return true;
    }
    
    @SubscribeEvent
    public void onClientTick(TickEvent.ClientTickEvent event)
    {
        if(event.phase == TickEvent.Phase.END && toOpen)
        {
            mc.addScheduledTask(() ->
            {
                mc.displayGuiScreen(new GuiEditHud());
                ChatUtils.printMessage("Opened Hud Editor");
                
                Backdoored.hudManager.elements.forEach(element -> ChatUtils.printMessage(element.getName() + " : " + element.getPos().toString()));
            });
            toOpen = false;
        }
    }
```

Reply:
I see.  I'm just going to say I disagree with the idea of using an event as a way to fix the issue.  We should try to come up with a more fluent workaround that allows other commands that may open screens to functional identically with a fix.

Reply:
Perhaps a potential fix would be to create a patch for that specific piece of code (GuiChat L132)
```Java
String s = this.inputField.getText().trim();

if (!s.isEmpty())
{
	this.sendChatMessage(s);
}

this.mc.displayGuiScreen((GuiScreen)null);
```

Perhaps like this:

```Java
String s = this.inputField.getText().trim();

if (!s.isEmpty())
{
	this.sendChatMessage(s);
}

if(mc.currentScreen == this)
{
	this.mc.displayGuiScreen((GuiScreen)null);
}
```